consensus/bor/bor.go

```diff
@@ -780,7 +784,18 @@ func (c *Bor) changeContractCodeIfNeeded(headerNumber uint64, state *state.State
 
 // FinalizeAndAssemble implements consensus.Engine, ensuring no uncles are set,
 // nor block rewards given, and returns the final block.
-func (c *Bor) FinalizeAndAssemble(chain consensus.ChainHeaderReader, header *types.Header, state *state.StateDB, txs []*types.Transaction, uncles []*types.Header, receipts []*types.Receipt) (*types.Block, error) {
+func (c *Bor) FinalizeAndAssemble(ctx context.Context, chain consensus.ChainHeaderReader, header *types.Header, state *state.StateDB, txs []*types.Transaction, uncles []*types.Header, receipts []*types.Receipt) (*types.Block, error) {
+
+	tracer := otel.GetTracerProvider().Tracer("MinerWorker")
+	finalizeCtx, finalizeSpan := tracer.Start(ctx, "FinalizeAndAssemble")
+	defer finalizeSpan.End()
+
+	finalizeSpan.SetAttributes(
+		attribute.Int("number", int(header.Number.Int64())),
+		attribute.String("hash", header.Hash().String()),
+		attribute.Int("number of txs", len(txs)),
```

Comment:
Could we add total gas used where it is possible?

Reply:
Sure, will add.